lib/pod/command/trunk/push.rb

```diff
@@ -119,6 +120,10 @@ def validate_podspec
           unless validator.validated?
             raise Informative, "The spec did not pass validation, due to #{validator.failure_reason}."
           end
+
+          # Let the validator's logic for the swift version
+          # set the value for the trunk JSON uploader
+          @swift_version = validator.swift_version
```

Comment:
Eh, I take it back, I might change this to use a new function

```
    # @return [String] A string representing the Swift version used during linting
    #                  or null, if Swift was not used.
    #
    def used_swift_version
      swift_version if @installer.pod_targets.any?(&:uses_swift?)
    end
```

Semantics aren't muddied then